File: npro/npro/report/previous_interviews_by_status/previous_interviews_by_status.py
# Copyright (c) 2013, GreyCube Technologies and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _
from frappe.utils import cint, today
import pandas as pd


def execute(filters=None):
    return get_data(filters)


def get_data(filters):
    filters["today"] = today()
    data = frappe.db.sql(
        """
            select 
                tja.status , tjo.customer_cf 
            from tabInterview ti 
            inner join `tabJob Opening` tjo on tjo.name = ti.job_opening 
            inner join `tabJob Applicant` tja on tja.name = ti.job_applicant 
        {where_conditions}
        """.format(
            where_conditions=get_conditions(filters),
        ),
        filters,
    )

    df1 = pd.DataFrame(data, columns=["status", "customer"])
    df2 = df1.pivot_table(index="customer", columns="status", aggfunc=len, fill_value=0)
    columns = [
        {
            "label": _("Customer"),
            "fieldname": "customer",
            "width": 180,
        },
    ] + [dict(label=d, fieldname=d, width=200) for d in df2.columns]

    return columns, df2.reset_index().to_dict(orient="records")


def get_conditions(filters):
    where_clause = []
    where_clause.append("ti.scheduled_on <= %(today)s")

    if filters.get("from_date"):
        where_clause.append("ti.scheduled_on >= %(from_date)s")

    if filters.get("to_date"):
        where_clause.append("ti.scheduled_on <= %(to_date)s")

    return " where " + " and ".join(where_clause) if where_clause else ""

File: npro/npro/report/previous_interviews_by_status/previous_interviews_by_status.js
// Copyright (c) 2016, GreyCube Technologies and contributors
// For license information, please see license.txt
/* eslint-disable */

frappe.query_reports["Previous Interviews by Status"] = {
  filters: [
    {
      fieldname: "from_date",
      label: __("Interview From Date"),
      fieldtype: "Date",
      default: moment().startOf("month"),
      reqd: 1,
    },
    {
      fieldname: "till_date",
      label: __("Interview Till Date"),
      fieldtype: "Date",
      default: moment(),
      reqd: 1,
    },
    {
      fieldname: "timespan",
      label: __("Interview Date in"),
      fieldtype: "Select",
      options: npro.utils.TIMESPAN_OPTIONS,
      on_change: function (query_report) {
        let date_range = npro.utils.get_date_range(
          query_report.get_values().timespan
        );
        frappe.query_report.set_filter_value({
          from_date: date_range[0],
          till_date: date_range[1],
        });
      },
      default: "This Month",
    },
  ],

  onload: function (report) {
    // report.page.set_title("");
  },
};
